RELEASE.md

```diff
@@ -2,6 +2,43 @@
 
 This document guides a contributor through creating a release of JupyterLab.
 
+## The JupyterLab Release Process
+
+JupyterLab follows [semver](https://semver.org/) for the versioning of the Python and JavaScript packages.
+
+### Release Timeline
+
+Although the commitments listed below are "best effort", the JupyterLab team tries to follow a couple of guidelines:
+
+- one major version per year, which usually includes API breaking changes
+- several minor versions per year that include new features but no API breaking changes
+- support and bug fixes on a couple of final releases (by backporting PRs and releasing from release branches)
+
+Release Plans are tracked in dedicated issues, and are closed when the final release. See the following two issues as an example:
+
+- [3.3 Release Plan](https://github.com/jupyterlab/jupyterlab/issues/11643)
+- [4.0 Release Plan](https://github.com/jupyterlab/jupyterlab/issues/9647)
+
+### Alpha Releases
+
+Alpha releases have a fairly low bar. Their purpose is to start putting the new JupyterLab version into the hands of users and extension authors.
+
+The requirements for an alpha release should be that JupyterLab can be installed and run. Bugs and breaking changes are accepted.
+
+### Beta Releases
+
+Beta releases usually try to not have breaking changes in the API, although breaking changes can sometimes happen during that phase if they were missed during the alpha stage.
+
+The recommended time period for the Beta phase is a minimum of 1 week.
```

Comment:
I would like to propose extending the beta testing period and including some form of a wider invitation for testing (a newsletter e-mail, a post on Discourse).
- we lightly touched upon an idea of sending out a survey to audit the UX changes at beta stage to catch any major regressions/issues which could be still fixed at this stage (in [a conversation]( https://github.com/jupyter/accessibility/issues/65#issuecomment-1030647307) with @isabela-pf). I am **not** suggesting to codify it here.
- it would be helpful to start working on a final changelog at this stage, as this is often something which gets omitted and users are often frustrated by this (our auto-generated changelogs are not especially user-friendly). Beta testers would also need that to understand what they should test out. Working on user-facing chagelog prior to final release is a common practice in other projects.

```suggestion
The recommended time period for the Beta phase is a minimum of 2 weeks.

The draft changelog describing user-facing changes will be published with the first Beta release.

The community of extension developers and active users will be invited to commence testing the new Beta release including the draft user-facing changelog, and an invitation to open issues for any major:
- regressions,
- usability problems
- points needing clarification (or inclusion) in the changelog, and
- points needing clarification in the extension porting guide.

The start of the Beta-testing period will be announced on Jupyter mailing group and Jupyter Discourse for major releases, and only via a Discourse post for minor releases.

All bug reports raised during the Beta-testing period should be triaged (but not necessarily addressed) before releasing the first release candidate.
```

Reply:
I second that if we want to more actively request user feedback (especially via survey) then two weeks is a more fitting timeline.

Would there be any aspects of release or community feelings people would anticipate from extending this?

Reply:
Thanks!

> as this is often something which gets omitted and users are often frustrated by this (our auto-generated changelogs are not especially user-friendly). Beta testers would also need that to understand what they should test out. Working on user-facing chagelog prior to final release is a common practice in other projects.

Yes this sounds good. Although most of the release process is now automated, we should probably still have a checklist that would list the steps to follow, such as reworking the automated changelog entries to make it more user friendly.